libpod/container_internal_freebsd.go

```diff
@@ -85,13 +85,36 @@ func (c *Container) prepare() error {
 	wg.Wait()
 
 	var createErr error
+	if createNetNSErr != nil {
+		createErr = createNetNSErr
+	}
 	if mountStorageErr != nil {
 		if createErr != nil {
 			logrus.Errorf("Preparing container %s: %v", c.ID(), createErr)
 		}
 		createErr = mountStorageErr
 	}
 
+	// Only trigger storage cleanup if mountStorage was successful.
+	// Otherwise, we may mess up mount counters.
+	if createNetNSErr != nil && mountStorageErr == nil {
```

Comment:
This would be easier to understand if the check was

if createErr != nil {

Reply:
This is the same logic as in container_internal_linux.go and I also found it confusing. I will try to rearrange it to make it easier to follow.